### PR TITLE
Powercube chain reformatting

### PR DESCRIPTION
**Pull request by destogl**
This is reformatting commit  discussed in #161. It is based on #161 so please merge it after.

orocos folder is not reformatted since ROS formatting is used and orocos files looked :ok:

### CODE REVIEW

File: schunk_powercube_chain/common/include/schunk_powercube_chain/Joint.h [hunk 1]
```diff
@@ -259,573 +266,562 @@ std::istream& operator>>(std::istream&, Joint<Real>& joint);
...
-%{
-	#include "Joint.h"
-%}
-%include "std_vector.i"
-%template(Jointd) Joint<double>;
 #endif
 
+#ifdef SWIG
+% module Util % include Source / Manipulation / ManipUtil / datastructsManipulator.h % include Source / Manipulation /
+    ManipUtil / Trajectory.h %
```
Comment (fmessmer) [inline schunk_powercube_chain/common/include/schunk_powercube_chain/Joint.h]: Don't know, but I think some wrong formatting happened here..

File: schunk_powercube_chain/common/include/schunk_powercube_chain/PowerCubeSim.h [hunk 2]
```diff
@@ -85,11 +84,11 @@ using namespace std;
 //#define __LINUX__
 
 #ifdef SWIG
-%module PowerCubeSim
-%include "Source/Manipulation/Interfaces/armInterface.h"
-%{
-	#include "PowerCubeSim.h"
-%}
+% module PowerCubeSim % include "Source/Manipulation/Interfaces/armInterface.h" %
```
Comment (fmessmer) [inline schunk_powercube_chain/common/include/schunk_powercube_chain/PowerCubeSim.h]: see comment above

File: schunk_powercube_chain/common/include/schunk_powercube_chain/datastructsManipulator.h [hunk 3]
```diff
@@ -67,82 +67,132 @@
...
 #ifdef SWIG
-%module PowerCubeCtrl
-%{
-    #include <schunk_powercube_chain/datastructsManipulator.h>
-%}
+% module PowerCubeCtrl %
+{
+#include <schunk_powercube_chain/datastructsManipulator.h>
+  %
+}
```
Comment (fmessmer) [inline schunk_powercube_chain/common/include/schunk_powercube_chain/datastructsManipulator.h]: see comment above

File: schunk_powercube_chain/common/include/schunk_powercube_chain/moveCommand.h [hunk 4]
```diff
@@ -66,122 +66,170 @@
...
-	 */
-	virtual double timeRemaining() { m_now.SetNow(); return (m_timeStarted - m_now) + getTotalTime(); }
+  moveCommand()
+  {
+    ;
+  }
+  virtual ~moveCommand()
+  {
+    ;
+  }
```
Comment (fmessmer) [inline schunk_powercube_chain/common/include/schunk_powercube_chain/moveCommand.h]: I assume, this was auto-formatted, right? Looked clearer before...

File: schunk_powercube_chain/common/include/schunk_powercube_chain/simulatedMotor.h [hunk 5]
```diff
@@ -65,97 +65,148 @@
...
-
-		double T0;					// Zeitkonstante für Annäherung der Sprungantwort durch Rampe
-
-		//ostream * deb;
+public:
+  simulatedMotor(double lowLimit, double upLimit, double maxAcc, double maxVel);
+  ~simulatedMotor()
+  {
+    ;
+  }
```
Comment (fmessmer) [inline schunk_powercube_chain/common/include/schunk_powercube_chain/simulatedMotor.h]: see comment above

**Comment by fmessmer** [comment]
In https://github.com/iirob/schunk_modular_robotics/blob/powercube_chain_reformatting/schunk_powercube_chain/common/src/PowerCubeCtrl.cpp, many occurences of something similar to:

```
-  std::cout << std::endl << "maxVel: ";
+  std::cout << std::endl
+            << "maxVel: ";
```

wer better readable before (without linebreak)...

This also somethimes occurs in strange line breaks within parameter lists of functions or mathematical formulae...

**Comment by fmessmer** [comment]
Other than my comments above, Visual Review :ok:
(I mainly reviewd it in https://github.com/ipa320/schunk_modular_robotics/pull/162/files?w=1 mode)

Has this branch been tested with hardware?

**Comment by destogl** [comment]
Yes It seamed something went wrong. I tested the code but since I don't have SWIG so I missed this error. 

To your last comment, yes I saw this and I agree but since I was ROS-Format I didn't change it.. Should I change it?

For me is not so easy to test it with hardware since I am using some stuff implemented in position_control branch... I can probably do it first tomorrow...

**Comment by destogl** [comment]
I found out why 

```
-  std::cout << std::endl << "maxVel: ";
+  std::cout << std::endl
+            << "maxVel: ";
```

happens. It is because `std::endl` is in the middle of the line which is IMHO at least questionable... Show I now also reformat his in e.g.:

```
-  std::cout << std::endl << "maxVel: ";
+  std::cout << std::endl;
+  std::cout << "maxVel: ";
```

Which make more sense?

**Comment by fmessmer** [comment]
Keeping it in a single line is fine, I guess
`std::cout << std::endl << "maxVel: ";`

**Comment by destogl** [comment]
OK. Than I think I solved issues from your comments. Tomorrow morning I test is on HW...

File: schunk_powercube_chain/common/src/PowerCubeSim.cpp [hunk 6]
```diff
@@ -66,697 +66,663 @@
...
+          deltaPhi = 0.5 * maxAccel * (n * deltaT * n * deltaT - (n - 1) * deltaT * (n - 1) * deltaT);
+          currVels[cubeNo] = maxAccel * simulatedTime;
+          std::cout << "Phase 1 ->";
+        }
+        else if ((simulatedTime > t) && (simulatedTime <= 2 * t))
+        {
+          // deltaPhi = maxVel *simulatedTime - 0.5*maxAccel*(simulatedTime -t)*(simulatedTime-t);
+          deltaPhi =
+              maxAccel * t * deltaT -
+              0.5 * maxAccel * ((n * deltaT - t) * (n * deltaT - t) - ((n - 1) * deltaT - t) * ((n - 1) * deltaT - t));
```
Comment (fmessmer) [inline schunk_powercube_chain/common/src/PowerCubeSim.cpp]: looks strange -> single line

File: schunk_powercube_chain/common/src/moveCommand.cpp [hunk 7]
```diff
@@ -331,156 +335,159 @@ double RampCommand::getTotalTime()
...
+        /// (System of equations would be overdetermined.)
+
+        /// Übersteuern, hier kann nicht gewährleistet werden, dass Abbremsphase
+        /// Während T3 (Gleichungssystem wäre überbestimmt).
+        /// out << "  Fall 4\n";
+        if (4 * delta * delta - 4 * delta * TG * v0 + 2 * TG * TG * v0 * v0 >= 0)
+          /// Root does not make any problems.
+          if (delta * v0 > 0)
+            acc = (-2.0 * delta + TG * v0 + sqrt(4 * delta * delta - 4 * delta * TG * v0 + 2 * TG * TG * v0 * v0)) /
+                  (TG * TG);
```
Comment (fmessmer) [inline schunk_powercube_chain/common/src/moveCommand.cpp]: looks strange -> single line

File: schunk_powercube_chain/common/src/moveCommand.cpp [hunk 8]
```diff
@@ -331,156 +335,159 @@ double RampCommand::getTotalTime()
...
+          else
+            acc = (-2.0 * delta + TG * v0 - sqrt(4 * delta * delta - 4 * delta * TG * v0 + 2 * TG * TG * v0 * v0)) /
+                  (TG * TG);
```
Comment (fmessmer) [inline schunk_powercube_chain/common/src/moveCommand.cpp]: looks strange -> single line

File: schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp [hunk 9]
```diff
@@ -136,20 +135,24 @@ class PowerCubeChainNode
     pc_ctrl_ = new PowerCubeCtrl(pc_params_);
 
     /// implementation of topics to publish
-    topicPub_JointState_ = n_.advertise<sensor_msgs::JointState> ("joint_states", 1);
-    topicPub_ControllerState_ = n_.advertise<control_msgs::JointTrajectoryControllerState> ("joint_trajectory_controller/state", 1);
+    topicPub_JointState_ = n_.advertise<sensor_msgs::JointState>("joint_states", 1);
+    topicPub_ControllerState_ =
+        n_.advertise<control_msgs::JointTrajectoryControllerState>("joint_trajectory_controller/state", 1);
```
Comment (fmessmer) [inline schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp]: looks strange -> single line

File: schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp [hunk 10]
```diff
@@ -136,20 +135,24 @@ class PowerCubeChainNode
...
     topicPub_Diagnostic_ = n_.advertise<diagnostic_msgs::DiagnosticArray>("diagnostics", 1);
 
     /// implementation of topics to subscribe
-    topicSub_CommandPos_ = n_.subscribe("joint_group_position_controller/command", 1, &PowerCubeChainNode::topicCallback_CommandPos, this);
-    topicSub_CommandVel_ = n_.subscribe("joint_group_velocity_controller/command", 1, &PowerCubeChainNode::topicCallback_CommandVel, this);
+    topicSub_CommandPos_ =
+        n_.subscribe("joint_group_position_controller/command", 1, &PowerCubeChainNode::topicCallback_CommandPos, this);
```
Comment (fmessmer) [inline schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp]: looks strange -> single line

File: schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp [hunk 11]
```diff
@@ -136,20 +135,24 @@ class PowerCubeChainNode
...
+    topicSub_CommandVel_ =
+        n_.subscribe("joint_group_velocity_controller/command", 1, &PowerCubeChainNode::topicCallback_CommandVel, this);
```
Comment (fmessmer) [inline schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp]: looks strange -> single line

File: schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp [hunk 12]
```diff
@@ -136,20 +135,24 @@ class PowerCubeChainNode
...
 
     /// implementation of service servers
     srvServer_Init_ = n_.advertiseService("driver/init", &PowerCubeChainNode::srvCallback_Init, this);
     srvServer_Stop_ = n_.advertiseService("driver/stop", &PowerCubeChainNode::srvCallback_Stop, this);
     srvServer_Recover_ = n_.advertiseService("driver/recover", &PowerCubeChainNode::srvCallback_Recover, this);
-    srvServer_SetOperationMode_ = n_.advertiseService("driver/set_operation_mode", &PowerCubeChainNode::srvCallback_SetOperationMode, this);
-    topicPub_OperationMode_ = n_.advertise<std_msgs::String> ("driver/current_operationmode", 1);
+    srvServer_SetOperationMode_ =
+        n_.advertiseService("driver/set_operation_mode", &PowerCubeChainNode::srvCallback_SetOperationMode, this);
```
Comment (fmessmer) [inline schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp]: looks strange -> single line

File: schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp [hunk 13]
```diff
@@ -632,38 +638,39 @@ class PowerCubeChainNode
...
 
     // set data to diagnostics
-    if(error_)
+    if (error_)
     {
       diagnostics.status[0].level = 2;
-      diagnostics.status[0].name = n_.getNamespace();;
-      diagnostics.status[0].message =  pc_ctrl_->getErrorMessage();
+      diagnostics.status[0].name = n_.getNamespace();
+      ;
```
Comment (fmessmer) [inline schunk_powercube_chain/ros/src/schunk_powercube_chain.cpp]: remove

**Comment by fmessmer** [comment]
There might be more "looks strange -> single line" in  schunk_powercube_chain/common/src/PowerCubeCtrl.cpp but the diff is not shown, because it's too many changes...

**Comment by destogl** [comment]
All lines you mentioned are corrected and I went through PowerCubeCtrl.cpp and corrected everything

**Comment by fmessmer** [comment]
Much appreciated!

**Pull request merged**